### PR TITLE
Feature: Removed margin when using the Compact Spacing option

### LINKED ISSUE #12012: Feature: Reduce margin in the compact spacing option
### What feature or improvement do you think would benefit Files?

Please add a small - compact view option for details layout (look at atached screenshot) 
image

Option to remove that big space between, for me now it's only one reason to not change my default explorer. Windows explorer it's more practical with this option
image

Option in original windows explorer
image

### Requirements

- None

### Files Version

2.4.61.0

### Windows Version

Windows 11 - 10.0.22621.1485

### Comments

_No response_

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #12012...

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Enable compact spacing
   2. Confirm there is no space between items in the details layout

**Screenshots (optional)**
image

### CODE REVIEW

File: src/Files.App/Services/ResourcesService.cs [hunk 1]
```diff
@@ -46,9 +46,11 @@ public void SetAppThemeFontFamily(string contentControlThemeFontFamily)
 		public void SetCompactSpacing(bool useCompactSpacing)
 		{
 			var listItemHeight = useCompactSpacing ? 24 : 36;
+			var listItemMargin = useCompactSpacing ? "-2" : "0";
```
Comment (yaira2) [inline src/Files.App/Services/ResourcesService.cs:49]: Is there a way to do this without the quotation marks?

**Comment by mdtauk** [comment]
This looks poorly designed

**Comment by ferrariofilippo** [comment]
> This looks poorly designed

Actually I love it
Anyway, @yaira2 I get this exception when I try to navigate to `Home`. The issue is that I fall in a crash loop and I can't start the app again
image

File: src/Files.App/Services/ResourcesService.cs (hunk 1 above)
Comment (ferrariofilippo) [reply src/Files.App/Services/ResourcesService.cs:49]: I think this is the best option in C#.
The alternative is using something like:
`var listItemMargin = 0 - (2 * (int)useCompactSpacing)`

**Comment by mdtauk** [comment]
I understand people wanting closer spacing, but it still needs some tweaks to make it look polished

**Comment by yaira2** [comment]
@ferrariofilippo can you check if the latest commit fixed the exception?

**Comment by yaira2** [comment]
> SecurityPage and AdvancedSecurityPage?

Do these need it? It should be using the default template now (compact styles don't apply to these pages anymore).

**Comment by ferrariofilippo** [comment]
> > SecurityPage and AdvancedSecurityPage?
> 
> Do these need it? It should be using the default template now (compact styles don't apply to these pages anymore).

I don't know why but I could still see the compact layout. Now it looks good!

> > RecentFiles
> 
> It's already applied to recent files, can you check again?

It's working. I guess `.txt` icons have some spacing above, since they are the only ones who doesn't seem centered

**Comment by yaira2** [comment]
Can you share a screenshot?

**Review by ferrariofilippo — APPROVED** [review]
LGTM!

**Comment by ferrariofilippo** [comment]
> Can you share a screenshot?




The txt file has 1/2 pixels of space above, instead the js file fills all the space

**Comment by yaira2** [comment]
This will be fixed when we switch to using smaller icons with compact spacing.

**Pull request merged**